specs/governance/governor.md

```diff
@@ -0,0 +1,702 @@
+
+```solidity
+function votableSupply() external view returns (uint256);
+```
+
+#### `votableSupply` (block)
+
+Returns the votable supply for `_blockNumber`.
+
+```solidity
+function votableSupply(uint256 _blockNumber) external view returns (uint256);
+```
+
+#### `votingDelay`
+
+Returns the delay since a proposal is submitted until voting power is fixed and voting starts.
+
+```solidity
+function votingDelay() external view returns (uint256);
+```
+
+#### `votingPeriod`
+
+Returns the delay since a proposal starts until voting ends.
+
+```solidity
+function votingPeriod() external view returns (uint256);
+```
+
+#### `weightCast`
+
+Returns the total number of votes that `_account` has cast for `_proposalId`.
+
+```solidity
+function weightCast(uint256 _proposalId, uint256 _account) external view returns (uint256);
+```
+
+### Events
+
+#### `ProposalCanceled`
+
+MUST trigger when a proposal is cancelled.
+
+```solidity
+event ProposalCanceled(uint256 proposalId);
+```
+
+#### `VoteCast`
+
+MUST trigger when proposal vote is casted.
+
+```solidity
+event VoteCast(address indexed voter, uint256 proposalId, uint8 support, uint256 weight, string reason);
+```
+
+#### `VoteCastWithParams`
+
+MUST trigger when proposal vote with parameters is casted.
+
+```solidity
+event VoteCastWithParams(address indexed voter, uint256 proposalId, uint8 support, uint256 weight, string reason, bytes params); 
+```
+
+#### `ProposalTypeUpdated`
+
+MUST trigger when a proposal type is updated.
+
+```solidity
+event ProposalTypeUpdated(uint256 indexed proposalId, uint8 proposalType);
+```
+
+#### `ProposalExecuted`
+
+MUST trigger when a proposal is executed.
+
+```solidity
+event ProposalExecuted(uint256 proposalId);
+```
+
+#### `ProposalCreated`
+
+MUST trigger when a proposal is created.
+
+```solidity
+event ProposalCreated(uint256 indexed proposalId, address indexed proposer, address indexed module, address[] targets, uint256[] values, string[] signatures, bytes[] calldatas, uint256 startBlock, uint256 endBlock, string description, uint8 proposalType);
+```
+
+#### `ProposalDeadlineUpdated`
+
+MUST trigger when a proposal's deadline is updated.
+
+```solidity
+event ProposalDeadlineUpdated(uint256 proposalId, uint64 deadline);
+```
+
+#### `ProposalThresholdSet`
+
+MUST trigger when the proposal threshold is updated.
+
+```solidity
+event ProposalThresholdSet(uint256 oldProposalThreshold, uint256 newProposalThreshold);
+```
+
+#### `VotingDelaySet`
+
+MUST trigger when the voting delay is updated.
+
+```solidity
+event VotingDelaySet(uint256 oldVotingDelay, uint256 newVotingDelay);
+```
+
+#### `VotingPeriodSet`
+
+MUST trigger when the voting period is updated.
+
+```solidity
+event VotingPeriodSet(uint256 oldVotingPeriod, uint256 newVotingPeriod);
+```
+
+#### `QuorumNumeratorUpdated`
+
+MUST trigger when the quorum numerator is updated.
+
+```solidity
+event QuorumNumeratorUpdated(uint256 oldQuorumNumerator, uint256 newQuorumNumerator);
+```
+
+## Storage
```

Comment:
We generally don't want to define storage as it's an implementation detail that doesn't impact the external interface. Goal of the spec is to define the behavior that is observable by external parties.

Reply:
gotcha, removed the `Storage` section in this case as the getter functions reflect the data that needs to be exposed. https://github.com/ethereum-optimism/specs/pull/315/commits/495398c73b5e5c4825f64b99864e99976a69f673